lib/Frontend/CompilerInvocation.cpp

```diff
@@ -659,7 +659,8 @@ static bool ParseDiagnosticArgs(DiagnosticOptions &Opts, ArgList &Args,
   Opts.SkipDiagnosticPasses |= Args.hasArg(OPT_disable_diagnostic_passes);
   Opts.ShowDiagnosticsAfterFatalError |=
     Args.hasArg(OPT_show_diagnostics_after_fatal);
-  Opts.UseColor |= Args.hasArg(OPT_color_diagnostics);
+  Opts.UseColor |= Args.hasFlag(OPT_color_diagnostics,
+                                OPT_no_color_diagnostics);
```

Comment:
At this point `llvm::sys::Process::StandardErrHasColors()` unconditionally returns `false`. Is this because this function is called in a subprocess?

Reply:
Yes, with the added fact that the parent process (the driver) is buffering output. It's still useful to have the default for people making frontend invocations directly, though.